tests/InstrEngineTests/NaglerInstrumentationMethod/NaglerInstrumentationMethod.cpp

```diff
@@ -747,21 +747,26 @@ HRESULT CInstrumentationMethod::OnModuleLoaded(_In_ IModuleInfo* pModuleInfo)
             CComPtr<IMetaDataImport2> pMetadataImport;
             IfFailRet(pModuleInfo->GetMetaDataImport((IUnknown**)&pMetadataImport));
 
+            BOOL bIsRejit = FALSE;
+            IfFailRet(pInstrumentMethodEntry->GetIsRejit(&bIsRejit));
 
-            MicrosoftInstrumentationEngine::CMetadataEnumCloser<IMetaDataImport2> spTypeDefEnum(pMetadataImport, nullptr);
-            mdTypeDef typeDef = mdTypeDefNil;
-            ULONG cTokens = 0;
-            while (S_OK == (hr = pMetadataImport->EnumTypeDefs(spTypeDefEnum.Get(), &typeDef, 1, &cTokens)))
+            if (bIsRejit)
```

Comment:
I'll add comments for this change. Our working theory is that .NET Core has an optimization where if ReJIT is called for a method before the JIT process has started, it will only callback to the profiler for ReJIT and not for the initial JIT (because the runtime already knows that initial JIT is not sufficient).

The test assembly used to blindly call ReJIT for each method the test cared about and would cause both the JIT and ReJIT callbacks to occur, however on .NET Core we only get the ReJIT callback. Thus, on .NET Core, if a test is expecting just the JIT, it will not get the expected callback and the test will fail.

The fix here is to only ask the method to be ReJIT'd if the test expects the callback for ReJIT.